apps/crn-server/src/handlers/teams/algolia-index-team-handler.ts

```diff
@@ -27,8 +26,25 @@ export const indexTeamHandler =
       logger.debug(`Fetched team ${teamId}`);
 
       if (team) {
+        const {
+          id,
+          displayName,
+          inactiveSince,
+          projectTitle,
+          labCount,
+          expertiseAndResourceTags,
+        } = team;
         await algoliaClient.save({
```

Comment:
i am not a big fan of the mapping above (or plucking properties really), it feels like we are constructing a new object just for algolia while previously it would have been the exact `team` object that comes out of the `fetchById` method (+ tags). The reason I am leaving my comment here is just for you to make sure this is type safe in a sense that - can you make sure that we are unable to save any extra properties or different properties. This still leaves more room for errors as people might come in and remap properties and its gonna be a mess but I would hope the types would at least give us some safety here

Reply:
I've added this `WithAlgoliaTag` here [review: guarantee team data type sent to algolia is correct](https://github.com/yldio/asap-hub/pull/3929/commits/951e3eb0412a623510ef662019b30880e23a9325) and removed type casting

So now if I add a property that is not required by team card (`TeamListItem`) I get a Typescript error

<img width="1419" alt="Screenshot 2023-11-30 at 10 50 22" src="https://github.com/yldio/asap-hub/assets/16595804/ed293b33-e6f5-4570-abb5-fe5dbb4404fb">


and if I remove a property required by TeamListItem + _tags I get a Typescript error as well

<img width="683" alt="Screenshot 2023-11-30 at 10 51 12" src="https://github.com/yldio/asap-hub/assets/16595804/d575408c-c648-4489-8077-378a7ece82a5">


I can add this `WithAlgoliaTag` to the rest of entities in `SavePayload` but I think it's better to do it in a separate PR